File: practice/algorithms/implementation/sequence_equation/SequenceEquation.js
'use strict';

const processData = input => {
    const lines = input.split('\n');
    const n = parseInt(lines[0]);
    const elements = lines[1].split(' ').map(i => parseInt(i));
    const result = [];
    for(let i = 1; i <= n; i++) {
        const index = elements.indexOf(i);
        const value1 = index + 1;
        const index2 = elements.indexOf(value1);
        result[i - 1] = index2 + 1;
    }

    console.log(result.join('\n'));
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/sequence_equation/SequenceEquation.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class SequenceEquation {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int tests = Integer.parseInt(stdin.nextLine());
        List<Integer> elements = Arrays.stream(stdin.nextLine().split(" ")).map(Integer::parseInt).collect(Collectors.toList());
        List<Integer> result = new ArrayList<>();
        for(int i = 1; i <= tests; i++) {
            int index = elements.indexOf(i);
            int value1 = index + 1;
            int index2 = elements.indexOf(value1);
            result.add(index2 + 1);
        }
        for (int elem : result) {
            System.out.println(elem);
        }
        stdin.close();
    }
}
